src/webgpu/util/math.ts

```diff
@@ -267,43 +267,42 @@ export function lerp(a: number, b: number, t: number) {
 }
 
 /** Unwrap Scalar params into numbers and check preconditions */
-function unwrapRangeParams(min: Scalar, max: Scalar, num_steps: Scalar) {
-  assert(min.type.kind === 'f32', '|min| needs to be a f32');
-  assert(max.type.kind === 'f32', '|max| needs to be a f32');
+function unwrapRangeParams(a: Scalar, b: Scalar, num_steps: Scalar) {
+  assert(a.type.kind === 'f32', '|a| needs to be a f32');
+  assert(b.type.kind === 'f32', '|b| needs to be a f32');
   assert(num_steps.type.kind === 'u32', '|num_steps| needs to be a u32');
 
-  const f32_min = min.value as number;
-  const f32_max = max.value as number;
+  const f32_a = a.value as number;
+  const f32_b = b.value as number;
   const u32_num_steps = num_steps.value as number;
 
-  assert(f32_max >= f32_min, '|max| must be greater than |min|');
   assert(u32_num_steps > 0, '|num_steps| must be greater than 0');
 
-  return { f32_min, f32_max, u32_num_steps };
+  return { f32_a, f32_b, u32_num_steps };
 }
 
 /** @returns a linear increasing range of numbers. */
-export function linearRange(min: Scalar, max: Scalar, num_steps: Scalar): Array<number> {
-  const { f32_min, f32_max, u32_num_steps } = unwrapRangeParams(min, max, num_steps);
+export function linearRange(a: Scalar, b: Scalar, num_steps: Scalar): Array<number> {
+  const { f32_a, f32_b, u32_num_steps } = unwrapRangeParams(a, b, num_steps);
 
   return Array.from(Array(u32_num_steps).keys()).map(i =>
-    lerp(f32_min, f32_max, i / (u32_num_steps - 1))
+    lerp(f32_a, f32_b, i / (u32_num_steps - 1))
   );
 }
 
 /**
- * @returns a non-linear increasing range of numbers, with a bias towards min.
+ * @returns a non-linear increasing range of numbers, with a bias towards the beginning.
  *
  * Generates a linear range on [0,1] with |num_steps|, then squares all the values to make the curve be quadratic,
  * thus biasing towards 0, but remaining on the [0, 1] range.
  * This biased range is then scaled to the desired range using lerp.
  * Different curves could be generated by changing c, where greater values of c will bias more towards 0.
  * */
-export function biasedRange(min: Scalar, max: Scalar, num_steps: Scalar): Array<number> {
+export function biasedRange(a: Scalar, b: Scalar, num_steps: Scalar): Array<number> {
```

Comment:
Why do these function use `Scalar` instead of just `number`?

Reply:
Probably legacy from me having rewritten it multiple times originally. Converted the code to use `number`